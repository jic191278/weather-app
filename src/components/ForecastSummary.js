import React from 'react';

import '../styles/forecast-summary.css';
import iconData from '../data/iconData.json';

const ForecastSummary = ({ forecast, handleForecastSelection }) => {
  const { date, description, icon, temperature } = forecast;
  const { max } = temperature;

  const iconCode = `${icon.slice(0, 1)}00`;
  const displayIcon = iconData[iconCode];

  const formattedDate = new Date(date).toDateString();

  return (
    <div className="forecast-summary">
      <p className="forcast-summary__date">{formattedDate}</p>
      <img
        className="forcast-summary__icon"
        src={displayIcon}
        data-testid="weather-icon"
        alt={`${description} icon`}
      />
      <p className="forcast-summary__temperature">{`${max}Â°c`}</p>
      <p className="forcast-summary__description">{description}</p>
      <button
        className="forcast-summary__button"
        onClick={handleForecastSelection(date)}
        type="submit"
      >
        More details
      </button>
    </div>
  );
};

export default ForecastSummary;
