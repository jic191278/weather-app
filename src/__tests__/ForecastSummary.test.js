import React from 'react';
import { render, screen } from '@testing-library/react';
import ForecastSummary from '../components/ForecastSummary';

describe('ForecastSummary', () => {
  const forecastProp = {
    date: 1710530097000, // Fri Mar 15 2024
    description: 'Crazy Weather',
    icon: '202',
    temperature: {
      max: 5,
      min: 1,
    },
  };

  describe('ForecastSummary Snapshot', () => {
    test('Component renders correctly', () => {
      const { asFragment } = render(
        <ForecastSummary forecast={forecastProp} />
      );

      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe('Component elements render', () => {
    beforeEach(() => {
      render(<ForecastSummary forecast={forecastProp} />);
    });

    test('Date displays correctly', () => {
      const dateParaElement = screen.getByText('Fri Mar 15 2024');
      expect(dateParaElement).toBeInTheDocument();
    });

    test('Weather icon is rendered', () => {
      const iconImgElement = screen.getByTestId('weather-icon');
      expect(iconImgElement).toHaveClass('forcast-summary__icon');
    });

    test('Max temperature is rendered', () => {
      const maxParaElement = screen.getByText('5Â°c');
      expect(maxParaElement).toBeInTheDocument();
    });

    test('Description is rendered', () => {
      const descParaElement = screen.getByText('Crazy Weather');
      expect(descParaElement).toBeInTheDocument();
    });

    test('More details button is rendered', () => {
      const button = screen.getByRole('button');
      expect(button).toBeInTheDocument();
    });
  });
});
