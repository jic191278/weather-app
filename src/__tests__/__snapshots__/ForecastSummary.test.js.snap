// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummary ForecastSummary Snapshot Component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-summary"
  >
    <p
      class="forcast-summary__date"
    >
      Fri Mar 15 2024
    </p>
    <img
      alt="Crazy Weather icon"
      class="forcast-summary__icon"
      data-testid="weather-icon"
      src="https://openweathermap.org/img/wn/11d@2x.png"
    />
    <p
      class="forcast-summary__temperature"
    >
      5Â°c
    </p>
    <p
      class="forcast-summary__description"
    >
      Crazy Weather
    </p>
    <button
      class="forcast-summary__button"
      name="more-details-btn"
      type="submit"
    >
      More details
    </button>
  </div>
</DocumentFragment>
`;
